### PR TITLE
Distribution of agents by percentage 

### PR DESCRIPTION
**Pull request by ChristianHirt**
Adds new Methods for Agent distribution:

distribute by percent and distribute till full inside a polygon

todo:

- [x] tests for Methods

### CODE REVIEW

**Review by JetteSchumann — CHANGES_REQUESTED** [review]

File: python_modules/jupedsim/jupedsim/distributions.py [hunk 1]
```diff
@@ -272,6 +272,54 @@ def distribute_in_circles_by_density(*, polygon, distance_to_agents, distance_to
...
+       fills the polygon entirely and then selects the percentage of placed agents
+
+        :param polygon: shapely polygon in which the agents will be placed
+        :param percent: percentage of agents selected - 100% ≙ completely filled polygon 0% ≙ 0 placed points
+        :param distance_to_agents: minimal distance between the centers of agents
+        :param distance_to_polygon: minimal distance between the center of agents and the polygon edges
+        :param seed: define a seed for random generation, Default value is None which corresponds to a random value
+        :param max_iterations: no more than max_iterations must find a point inside the polygon, Default is 10_000
+        :return: list of created points"""
+    samples = distribute_til_full(polygon=polygon, distance_to_agents=distance_to_agents,
```
[the remark this answers is not in the document]
Comment (ChristianHirt) [reply python_modules/jupedsim/jupedsim/distributions.py:316]: resolved as discussed

File: python_modules/jupedsim/jupedsim/distributions_app.py [hunk 2]
```diff
@@ -202,6 +202,45 @@ def main():
...
+                "find another place to put an agent before a polygon is considered filled\n"
+                "General settings can be made in the left sidebar")
+        max_iterations = st.number_input("maximum tries to find a new point", 1, value=10_000, step=100)
+        button_clicked = st.button('distribute agents')
+
+        if button_clicked:
+            samples = distributions.distribute_til_full(polygon=s_polygon, distance_to_agents=distance_to_agents,
+                                                        distance_to_polygon=distance_to_polygon, seed=seed,
+                                                        max_iterations=max_iterations)
+            st.text('Below should be a plot')
```
[the remark this answers is not in the document]
Comment (ChristianHirt) [reply python_modules/jupedsim/jupedsim/distributions_app.py]: added

File: python_modules/jupedsim/jupedsim/distributions_app.py [hunk 3]
```diff
@@ -202,6 +202,45 @@ def main():
         else:
             show_points(s_polygon=s_polygon, samples=[], radius=distance_to_agents, obstacles=obstacles)
 
+    if distribution_type == "full":
+        st.text("Instructions:\n"
+                "distributes agents all over the polygon\n"
+                "you can enter how many tries there should be to\n"
```
[the remark this answers is not in the document]
Comment (ChristianHirt) [reply python_modules/jupedsim/jupedsim/distributions_app.py]: removed

File: python_modules/jupedsim/jupedsim/distributions_app.py [hunk 4]
```diff
@@ -202,6 +202,45 @@ def main():
         else:
             show_points(s_polygon=s_polygon, samples=[], radius=distance_to_agents, obstacles=obstacles)
 
+    if distribution_type == "full":
+        st.text("Instructions:\n"
+                "distributes agents all over the polygon\n"
```
[the remark this answers is not in the document]
Comment (ChristianHirt) [reply python_modules/jupedsim/jupedsim/distributions_app.py]: instructions rewritten

File: python_modules/jupedsim/jupedsim/distributions.py [hunk 5]
```diff
@@ -272,6 +272,54 @@ def distribute_in_circles_by_density(*, polygon, distance_to_agents, distance_to
                                            numbers_of_agents=number_of_agents, seed=seed, max_iterations=max_iterations)
```
[the remark this answers is not in the document]
Comment (ChristianHirt) [reply python_modules/jupedsim/jupedsim/distributions.py:274]: readme updated

File: python_modules/jupedsim/jupedsim/distributions_app.py [hunk 6]
```diff
@@ -202,6 +202,45 @@ def main():
...
+                "you can enter how many tries there should be to\n"
+                "find another place to put an agent before a polygon is considered filled\n"
+                "General settings can be made in the left sidebar")
```
[the remark this answers is not in the document]
Comment (ChristianHirt) [reply python_modules/jupedsim/jupedsim/distributions_app.py:215]: general instructions are explained in sidebar
specific informations are explained after selection of distribution type

**Review by JetteSchumann — CHANGES_REQUESTED** [review]

**Review by JetteSchumann — APPROVED** [review]

**Pull request merged**